sdk/python/foundation-models/system/content-safety/deploy_with_content_safety.ipynb

```diff
@@ -0,0 +1,803 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# How to create an Safety-Enabled LLaMa Online Endpoint\n",
+    "## This notebook will walk you through the steps to create a Safety-Enabled LLaMa Online Endpoint.\n",
+    "### The steps are:\n",
+    "1. Create a Content Safety for moderating the request from user and response from the LLaMa Online Endpoint\n",
+    "2. Create a new LLaMa Online Endpoint\n",
+    "3. Create a new Safety-Enabled LLaMa Online Endpoint with a custom score.py file which will be used to moderate the request and response"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1. Prerequisites\n",
+    "#### 1.1 Install Dependencies"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install azure-identity==1.14.0b1\n",
+    "%pip install azure-mgmt-cognitiveservices==13.4.0\n",
+    "%pip install --pre azure-ai-ml \n",
+    "%pip install --pre azure-mgmt-msi\n",
+    "%pip install --pre azure-mgmt-authorization"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 1.2 Assign variables for the workspace and deployment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "subscription_id = \"<SUBSCRIPTION_ID>\"\n",
+    "resource_group = \"<RESOURCE_GROUP>\"\n",
+    "workspace_name = \"<AML_WORKSPACE_NAME>\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 1.3 Decide on a name for your Safety LLaMa Online Endpoint"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# import random\n",
+    "# rand = random.randint(0, 10000)\n",
+    "# endpoint_name = f\"safetyllama{rand}\" \n",
+    "endpoint_name = \"<ONLINE-ENDPOINT-NAME>\" # the final endpoint name of the safety enabled llama endpoint\n",
```

Comment:
I would set the random name as default, and put a comment for the user to change it if they want, calling out that the endpoint name needs to be unique

Reply:
Maybe a side topic, will the name of the three resources - LLaMA endpoint, AACS endpoint, AACS-enabled-LLaMA endpoint , have a common prefix? I believe with a common prefix, it's more friendly to let customer understand they are connected.

Reply:
yes,  with the given name, there will be 4 resources created in the resource group of your AML workspace:
- 1. an Azure AI Content Safety resource: {endpoint_name}-aacs
- 2. a LLaMa online endpoint: {endpoint_name}-llama
- 3. a UAI(User Assigned Identity): {endpoint_name}-uia
- 4. an Azure AI Content Safety enabled LLaMa online endpoint, which you will be using it to do your AI work: {endpoint_name}